### PR TITLE
Write all rz components

### PR DESCRIPTION
**Pull request by dpgrote**
This PR writes all of the RZ field components to the plot files. The quantities "Ex" etc are still the total fields at theta = 0 (sum of the real components). The other components are written out with names likes Ex_1_real and Jz_2_imag.

### CODE REVIEW

File: Source/Diagnostics/FieldIO.cpp [hunk 1]
```diff
@@ -352,6 +352,31 @@ AverageAndPackVectorField( MultiFab& mf_avg,
...
+                                     const DistributionMapping& dm,
+                                     int& dcomp, const int ngrow )
+{
+    if (vector_field[0]->nComp() > 1) {
+        std::array<std::unique_ptr<MultiFab>,3> vector_field_component;
+        vector_field_component[0].reset(new MultiFab(vector_field[0]->boxArray(), dm, 1, vector_field[0]->nGrowVect()));
+        vector_field_component[1].reset(new MultiFab(vector_field[1]->boxArray(), dm, 1, vector_field[1]->nGrowVect()));
+        vector_field_component[2].reset(new MultiFab(vector_field[2]->boxArray(), dm, 1, vector_field[2]->nGrowVect()));
+        for (int icomp=0 ; icomp < vector_field[0]->nComp() ; icomp++) {
+            // This is a bit inefficient since it copies the data. Can this be done with pointers?
```
Comment (ax3l) [inline Source/Diagnostics/FieldIO.cpp]: maybe @WeiqunZhang knows this.

Comment (WeiqunZhang) [reply Source/Diagnostics/FieldIO.cpp]: You can do

```
vector_field_component[0].reset(new MultiFab(*vector_field[0], amrex::make_alias, icomp, 1));
..[1]...
...[2]...
AverageAndPackVectorField(...)
```
without deep copy.

Comment (ax3l) [reply Source/Diagnostics/FieldIO.cpp]: @RemiLehe @dpgrote did you see this comment?

Comment (dpgrote) [reply Source/Diagnostics/FieldIO.cpp]: Yep. I'll make the change.

**Pull request merged**